ERROR - 2025-06-30 14:03:40 --> Error connecting to the database: TypeError: str_replace(): Argument #1 ($search) must be of type array|string, null given in C:\xampp\htdocs\churchapp\system\Database\MySQLi\Connection.php:244
Stack trace:
#0 C:\xampp\htdocs\churchapp\system\Database\MySQLi\Connection.php(244): str_replace(NULL, '****', 'No connection c...')
#1 C:\xampp\htdocs\churchapp\system\Database\BaseConnection.php(421): CodeIgniter\Database\MySQLi\Connection->connect(false)
#2 C:\xampp\htdocs\churchapp\system\Database\BaseConnection.php(620): CodeIgniter\Database\BaseConnection->initialize()
#3 C:\xampp\htdocs\churchapp\system\Database\BaseBuilder.php(1649): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#4 C:\xampp\htdocs\churchapp\system\Model.php(286): CodeIgniter\Database\BaseBuilder->get()
#5 C:\xampp\htdocs\churchapp\system\BaseModel.php(675): CodeIgniter\Model->doFindAll(0, 0)
#6 C:\xampp\htdocs\churchapp\app\Controllers\FamilyController.php(411): CodeIgniter\BaseModel->findAll()
#7 C:\xampp\htdocs\churchapp\system\CodeIgniter.php(933): App\Controllers\FamilyController->donationPurposes()
#8 C:\xampp\htdocs\churchapp\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\FamilyController))
#9 C:\xampp\htdocs\churchapp\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#10 C:\xampp\htdocs\churchapp\system\Boot.php(334): CodeIgniter\CodeIgniter->run()
#11 C:\xampp\htdocs\churchapp\system\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#12 C:\xampp\htdocs\churchapp\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#13 {main}
CRITICAL - 2025-06-30 14:03:40 --> CodeIgniter\Database\Exceptions\DatabaseException: Unable to connect to the database.
Main connection [MySQLi]: str_replace(): Argument #1 ($search) must be of type array|string, null given
[Method: GET, Route: donationpurposes]
in SYSTEMPATH\Database\BaseConnection.php on line 467.
 1 SYSTEMPATH\Database\BaseConnection.php(620): CodeIgniter\Database\BaseConnection->initialize()
 2 SYSTEMPATH\Database\BaseBuilder.php(1649): CodeIgniter\Database\BaseConnection->query('SELECT *
FROM `donation_purposes`', [], false)
 3 SYSTEMPATH\Model.php(286): CodeIgniter\Database\BaseBuilder->get()
 4 SYSTEMPATH\BaseModel.php(675): CodeIgniter\Model->doFindAll(0, 0)
 5 APPPATH\Controllers\FamilyController.php(411): CodeIgniter\BaseModel->findAll()
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\FamilyController->donationPurposes()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\FamilyController))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(334): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-06-30 14:13:49 --> CodeIgniter\View\Exceptions\ViewException: Invalid file: "family/payment.php"
[Method: GET, Route: payment]
in SYSTEMPATH\Exceptions\FrameworkException.php on line 39.
 1 SYSTEMPATH\View\View.php(212): CodeIgniter\Exceptions\FrameworkException::forInvalidFile('family/payment.php')
 2 SYSTEMPATH\Common.php(1173): CodeIgniter\View\View->render('family/payment', [], true)
 3 APPPATH\Controllers\FamilyController.php(473): view('family/payment', [...])
 4 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\FamilyController->payment()
 5 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\FamilyController))
 6 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 7 SYSTEMPATH\Boot.php(334): CodeIgniter\CodeIgniter->run()
 8 SYSTEMPATH\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 9 FCPATH\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
ERROR - 2025-06-30 14:22:46 --> mysqli_sql_exception: Column 'family_id' cannot be null in C:\xampp\htdocs\churchapp\system\Database\MySQLi\Connection.php:327
Stack trace:
#0 C:\xampp\htdocs\churchapp\system\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `do...', 0)
#1 C:\xampp\htdocs\churchapp\system\Database\BaseConnection.php(738): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `do...')
#2 C:\xampp\htdocs\churchapp\system\Database\BaseConnection.php(652): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `do...')
#3 C:\xampp\htdocs\churchapp\system\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `do...', Array, false)
#4 C:\xampp\htdocs\churchapp\system\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
#5 C:\xampp\htdocs\churchapp\system\BaseModel.php(839): CodeIgniter\Model->doInsert(Array)
#6 C:\xampp\htdocs\churchapp\system\Model.php(800): CodeIgniter\BaseModel->insert(Array, false)
#7 C:\xampp\htdocs\churchapp\system\BaseModel.php(750): CodeIgniter\Model->insert(Array, false)
#8 C:\xampp\htdocs\churchapp\app\Controllers\FamilyController.php(488): CodeIgniter\BaseModel->save(Array)
#9 C:\xampp\htdocs\churchapp\system\CodeIgniter.php(933): App\Controllers\FamilyController->storeDonation()
#10 C:\xampp\htdocs\churchapp\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\FamilyController))
#11 C:\xampp\htdocs\churchapp\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#12 C:\xampp\htdocs\churchapp\system\Boot.php(334): CodeIgniter\CodeIgniter->run()
#13 C:\xampp\htdocs\churchapp\system\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#14 C:\xampp\htdocs\churchapp\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#15 {main}
CRITICAL - 2025-06-30 14:22:46 --> CodeIgniter\Database\Exceptions\DatabaseException: Column 'family_id' cannot be null
[Method: POST, Route: donations/store]
in SYSTEMPATH\Database\BaseConnection.php on line 692.
 1 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `donations` (`family_id`, `purpose_id`, `amount`, `donation_date`, `notes`, `created_at`, `updated_at`) VALUES (:family_id:, :purpose_id:, :amount:, :donation_date:, :notes:, :created_at:, :updated_at:)', [...], false)
 2 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 3 SYSTEMPATH\BaseModel.php(839): CodeIgniter\Model->doInsert([...])
 4 SYSTEMPATH\Model.php(800): CodeIgniter\BaseModel->insert([...], false)
 5 SYSTEMPATH\BaseModel.php(750): CodeIgniter\Model->insert([...], false)
 6 APPPATH\Controllers\FamilyController.php(488): CodeIgniter\BaseModel->save([...])
 7 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\FamilyController->storeDonation()
 8 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\FamilyController))
 9 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
10 SYSTEMPATH\Boot.php(334): CodeIgniter\CodeIgniter->run()
11 SYSTEMPATH\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
12 FCPATH\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-06-30 14:22:46 --> [Caused by] CodeIgniter\Database\Exceptions\DatabaseException: Column 'family_id' cannot be null
in SYSTEMPATH\Database\MySQLi\Connection.php on line 332.
 1 SYSTEMPATH\Database\BaseConnection.php(738): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `donations` (`family_id`, `purpose_id`, `amount`, `donation_date`, `notes`, `created_at`, `updated_at`) VALUES (NULL, \'1\', \'2\', \'2025-07-04\', \'\', \'2025-06-30 14:22:46\', \'2025-06-30 14:22:46\')')
 2 SYSTEMPATH\Database\BaseConnection.php(652): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `donations` (`family_id`, `purpose_id`, `amount`, `donation_date`, `notes`, `created_at`, `updated_at`) VALUES (NULL, \'1\', \'2\', \'2025-07-04\', \'\', \'2025-06-30 14:22:46\', \'2025-06-30 14:22:46\')')
 3 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `donations` (`family_id`, `purpose_id`, `amount`, `donation_date`, `notes`, `created_at`, `updated_at`) VALUES (:family_id:, :purpose_id:, :amount:, :donation_date:, :notes:, :created_at:, :updated_at:)', [...], false)
 4 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 5 SYSTEMPATH\BaseModel.php(839): CodeIgniter\Model->doInsert([...])
 6 SYSTEMPATH\Model.php(800): CodeIgniter\BaseModel->insert([...], false)
 7 SYSTEMPATH\BaseModel.php(750): CodeIgniter\Model->insert([...], false)
 8 APPPATH\Controllers\FamilyController.php(488): CodeIgniter\BaseModel->save([...])
 9 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\FamilyController->storeDonation()
10 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\FamilyController))
11 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 SYSTEMPATH\Boot.php(334): CodeIgniter\CodeIgniter->run()
13 SYSTEMPATH\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
14 FCPATH\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-06-30 14:22:46 --> [Caused by] mysqli_sql_exception: Column 'family_id' cannot be null
in SYSTEMPATH\Database\MySQLi\Connection.php on line 327.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `donations` (`family_id`, `purpose_id`, `amount`, `donation_date`, `notes`, `created_at`, `updated_at`) VALUES (NULL, \'1\', \'2\', \'2025-07-04\', \'\', \'2025-06-30 14:22:46\', \'2025-06-30 14:22:46\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(738): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `donations` (`family_id`, `purpose_id`, `amount`, `donation_date`, `notes`, `created_at`, `updated_at`) VALUES (NULL, \'1\', \'2\', \'2025-07-04\', \'\', \'2025-06-30 14:22:46\', \'2025-06-30 14:22:46\')')
 3 SYSTEMPATH\Database\BaseConnection.php(652): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `donations` (`family_id`, `purpose_id`, `amount`, `donation_date`, `notes`, `created_at`, `updated_at`) VALUES (NULL, \'1\', \'2\', \'2025-07-04\', \'\', \'2025-06-30 14:22:46\', \'2025-06-30 14:22:46\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `donations` (`family_id`, `purpose_id`, `amount`, `donation_date`, `notes`, `created_at`, `updated_at`) VALUES (:family_id:, :purpose_id:, :amount:, :donation_date:, :notes:, :created_at:, :updated_at:)', [...], false)
 5 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 6 SYSTEMPATH\BaseModel.php(839): CodeIgniter\Model->doInsert([...])
 7 SYSTEMPATH\Model.php(800): CodeIgniter\BaseModel->insert([...], false)
 8 SYSTEMPATH\BaseModel.php(750): CodeIgniter\Model->insert([...], false)
 9 APPPATH\Controllers\FamilyController.php(488): CodeIgniter\BaseModel->save([...])
10 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\FamilyController->storeDonation()
11 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\FamilyController))
12 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
13 SYSTEMPATH\Boot.php(334): CodeIgniter\CodeIgniter->run()
14 SYSTEMPATH\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
15 FCPATH\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
